util/core_local.h

```diff
@@ -38,23 +38,23 @@ class CoreLocalArray {
 
  private:
   std::unique_ptr<T[]> data_;
-  size_t size_shift_;
+  int size_shift_;
 };
 
 template<typename T>
 CoreLocalArray<T>::CoreLocalArray() {
-  unsigned int num_cpus = std::thread::hardware_concurrency();
+  int num_cpus = static_cast<int>(std::thread::hardware_concurrency());
   // find a power of two >= num_cpus and >= 8
   size_shift_ = 3;
-  while (1u << size_shift_ < num_cpus) {
+  while (1 << size_shift_ < num_cpus) {
```

Comment:
also, unbelievably, the array index on line 52 has to be a size_t to avoid the warning about bitshifting+implicit 32-to-64-bit conversion. maybe we should just disable that warning, it makes dealing with arrays + bitshifting way too annoying.